src/main/scala/cromwell/engine/backend/local/SharedFileSystem.scala

```diff
@@ -130,11 +137,30 @@ trait SharedFileSystem {
       WdlFile(WdlFile.appendPathsWithSlashSeparators("/root", pathWithWorkflowName.subpath(1, pathWithWorkflowName.getNameCount).toString))
     }
 
-    // If this call is using Docker, adjust input paths, otherwise return unaltered input paths.
+    def localizeInputToCallDirectory(path: String): Path = {
+      val originalPath = Paths.get(path)
+      val callPath = Try {
+        val callInputsDirectory = LocalBackend.hostCallPath(workflowDescriptor, call.name, callKey.index).resolve("inputs")
+        if (!callInputsDirectory.toFile.exists) callInputsDirectory.toFile.mkdirs
+        val hash = hashPath(originalPath)
+        callInputsDirectory.resolve(s"$hash-${originalPath.getFileName.toString}")
+      }
+
+      callPath.toOption flatMap { cp =>
+        CallLocalizers map { _(Option(callKey.scope), originalPath, cp) } find { _.isSuccess } map { _ => cp }
+      } getOrElse {
+        logger.warn(s"Unable to isolate call inputs for call ${call.name}")
+        originalPath
+      }
+    }
+
+    // Try to localize input files to the call directory. If it's docker call, also adjust input path to container pathing
     def adjustPath(nameAndValue: (String, WdlValue)): (String, WdlValue) = {
       val (name, value) = nameAndValue
       val adjusted = value match {
-        case WdlFile(path) => containerPath(path)
+        case WdlFile(path) =>
+          val localizedPath = localizeInputToCallDirectory(path).toAbsolutePath.toString
+          if (call.docker.isDefined) containerPath(localizedPath) else WdlFile(localizedPath)
```

Comment:
Hmm not 100% sure what you mean :)
There is still `call.docker map { _ => containerPath(localizedPath) } getOrElse WdlFile(localizedPath)` if you think that's better.
But I prefer the if/else because `call.docker map ...` is misleading as we don't really "map" the docker value.